src/openrct2/actions/CheatSetAction.cpp

```diff
@@ -512,13 +512,23 @@ void CheatSetAction::FixBrokenRides() const
 {
     for (auto& ride : GetRideManager())
     {
-        if ((ride.mechanic_status != RIDE_MECHANIC_STATUS_FIXING)
-            && (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN)))
+        if (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN))
         {
             auto mechanic = RideGetAssignedMechanic(ride);
-            if (mechanic != nullptr)
+
+            if (ride.mechanic_status == RIDE_MECHANIC_STATUS_FIXING)
             {
-                mechanic->RemoveFromRide();
+                if (mechanic != nullptr)
+                {
+                    mechanic->RideSubState = PeepRideSubState::ApproachExitWaypoints;
+                }
```

Comment:
Its probably best to check if mechanic is nullptr after line 517 and continue the loop if it is null.